test/include/mbgl/test/vector_tile_test.hpp

```diff
@@ -38,6 +38,12 @@ class VectorTileTest {
                                   imageManager,
                                   glyphManager,
                                   0};
+
+    ~VectorTileTest() {
+        // Ensure that deferred releases are complete before cleaning up
+        EXPECT_EQ(0, loop.waitForEmpty(std::chrono::milliseconds{100}));
```

Comment:
Having a time based condition here in a test isn't ideal, this would be a spot to either use an infinite timeout or a much longer delay.

Reply:
I just made it an unlimited timeout under the assumption that the unit test process has a timeout that would handle any deadlock reasonably.